### PR TITLE
Migrate db.js to TS with unit tests.

### PR DESCRIPTION
**Pull request by hiyaryan**
This PR migrates db.js to TS, db.ts, and adds unit tests. The follow changes have been made

- Updates import path in `app.js`.
- Adds types to `connectDB` function signature.
- Adds error handling to `connectDB` for undefined env variables. Adds unit tests with 94.11% coverage.
  - Unit test using production env variables is skipped to ensure consistency across development environments.

All unit tests pass using `npm run test`. The application runs in the container as expected and connects to the corresponding environment database using `db.ts`.

### CODE REVIEW

File: backend/tests/db.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    expect(mongoose.connect).toHaveBeenCalledWith(`${mongoUri}/testdb`);
+    expect(console.log).toHaveBeenCalledWith(`Connected to local MongoDB: ${mongoUri}/testdb`);
+  });
+
+  it('should connect to local test MongoDB in test environment', async () => {
+    process.env.NODE_ENV = 'test';
+
+    await connectDB('testdb');
+
+    expect(mongoose.connect).toHaveBeenCalledWith('mongodb://localhost:27017/testdb-test');
```
Comment (davidzlu) [inline backend/tests/db.test.ts:56]: Not a change request, but something to think about for the future is what the test environment will be. Will all unit tests only use the memory mongodb? Will the test environment have a dedicated test db, and will it always be in the same host as the test runner? Will everything be in a container and therefore have different host names?

**Review by davidzlu — CHANGES_REQUESTED** [review]
It's looking good so far, and the tests seem comprehensive! There's an error that appears only when you run the db tests alone, and I think the core issue is how to setup the memory mongodb and manage a mongoose connection across tests.

File: backend/src/app.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { access, entry } from './routes/index.js';
 import ExpressError from './utils/ExpressError.js';
 import User from './models/user.js';
 
-import connectDB from './db.js';
+import connectDB from './db';
```
[the remark this answers is not in the document]
Comment (hiyaryan) [reply backend/src/app.js:6]: I kept a copy of db.js locally after I converted db.ts. So to ensure that it used db.ts, I removed the file extension. That's probably the only reason to remove the file extension. You think we should keep the file extensions for TypeScript files as `.js`?

File: backend/tests/db.test.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable sort-imports */
+
+import dotenv from 'dotenv';
+dotenv.config();
```
[the remark this answers is not in the document]
Comment (hiyaryan) [reply backend/tests/db.test.ts:5]: Will update! Much nicer one liner.

File: backend/tests/db.test.ts
[the remark this answers is not in the document]
Comment (hiyaryan) [reply backend/tests/db.test.ts:1]: Are you still getting this error after fast-forwarding this PR with your updates from your `utils` PR?

File: backend/tests/db.test.ts (hunk 1 above)
Comment (hiyaryan) [reply backend/tests/db.test.ts:56]: Great questions! For the record, we decided in @davidzlu [`utils`](#158 ) PR to have standalone memory mongodbs for each test suite that needs one. So we will not be using one database for the entire test environment which could lead to unexpected test results.

**Review by davidzlu — APPROVED** [review]
Great job! It all looks good!

File: backend/src/app.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidzlu) [reply backend/src/app.js]: I think it's fine I was just wondering

**Pull request merged**